src/base/BasePaymaster.sol

```diff
@@ -12,11 +12,12 @@ import { IEntryPoint } from "@account-abstraction-v7/interfaces/IEntryPoint.sol"
  * provides helper methods for staking.
  * Validates that the postOp is called only by the entryPoint.
  */
-abstract contract BasePaymaster is Ownable {
+abstract contract BasePaymaster is AccessControl {
     IEntryPoint public immutable entryPoint;
 
-    constructor(address _entryPoint, address _owner) Ownable(_owner) {
+    constructor(address _entryPoint, address _owner) {
         entryPoint = IEntryPoint(_entryPoint);
+        _grantRole(DEFAULT_ADMIN_ROLE, _owner);
```

Comment:
could we also set the manager in the constructor just so that we don't have to send another tx after deployment